test/performance/odp_pool_latency.c

```diff
@@ -0,0 +1,1381 @@
+				/* Due to potential pattern errors, there might not be expected
+				 * amount of freeable events. */
+				val = ODPH_MIN(val, cur_idx - head_idx);
+
+				if (elem->opt == TOP) {
+					idx = head_idx;
+					head_idx += val;
+				} else {
+					cur_idx -= val;
+					idx = cur_idx;
+				}
+
+				free_fn(data, idx, val, stats, i, j, is_saved);
+			} else {
+				odp_time_wait_ns(val);
+			}
+		}
+	}
+
+	t2 = odp_time_local_strict();
+	stats->tot_tm = odp_time_diff_ns(t2, t1);
+	odp_barrier_wait(&config->prog_config->term_barrier);
+
+	return 0;
+}
+
+static odp_bool_t setup_workers(prog_config_t *config)
+{
+	odph_thread_common_param_t thr_common;
+	odph_thread_param_t thr_params[config->num_workers], *thr_param;
+
+	odp_barrier_init(&config->init_barrier, config->num_workers + 1);
+	odp_barrier_init(&config->term_barrier, config->num_workers + 1);
+	odph_thread_common_param_init(&thr_common);
+	thr_common.instance = config->odp_instance;
+	thr_common.cpumask = &config->worker_mask;
+
+	for (uint32_t i = 0; i < config->num_workers; ++i) {
+		thr_param = &thr_params[i];
+		odph_thread_param_init(thr_param);
+		thr_param->start = run_test;
+		thr_param->thr_type = ODP_THREAD_WORKER;
+		thr_param->arg = &config->worker_config[i];
+	}
+
+	if ((uint32_t)odph_thread_create(config->thread_tbl, &thr_common, thr_params,
+					 config->num_workers) != config->num_workers) {
+		ODPH_ERR("Error configuring worker threads\n");
+		return false;
+	}
+
+	odp_barrier_wait(&config->init_barrier);
+
+	return true;
+}
+
+static odp_bool_t setup_test(prog_config_t *config)
+{
+	return setup_worker_config(config) && setup_workers(config);
+}
+
+static void stop_test(prog_config_t *config)
+{
+	odp_barrier_wait(&config->term_barrier);
+	(void)odph_thread_join(config->thread_tbl, config->num_workers);
+}
+
+static void print_stats(const prog_config_t *config)
+{
+	const alloc_elem_t *elem;
+	const stats_t *stats;
+	uint64_t ev_rate, ave_b_alloc_tm, b_alloc_min, b_alloc_max, ave_b_free_tm, b_free_min,
+	b_free_max, ave_alloc_tm, ave_free_tm, ave_ua_b_tm, b_ua_min, b_ua_max, ave_ua_tm,
+	tot_b_alloc_tm = 0U, tot_b_free_tm = 0U, tot_alloc_tm = 0U, tot_free_tm = 0U,
+	tot_alloc_min = 0U, tot_alloc_max = 0U, tot_free_min = 0U, tot_free_max = 0U,
+	tot_b_ua_tm = 0U, tot_ua_tm = 0U, tot_ua_min = 0U, tot_ua_max = 0U;
+
+	printf("\n==================\n\n"
+	       "Pool latency test done\n\n"
+	       "    type:         %s\n"
+	       "    event count:  %u\n", config->type == BUFFER ? "buffer" :
+		config->type == PACKET ? "packet" : config->type == TMO ? "timeout" : "vector",
+	       config->num_evs);
+
+	if (config->type != TMO)
+		printf("    %s  %u\n", config->type != VECTOR ? "data size:  " : "vector size:",
+		       config->data_size);
+
+	printf("    pool policy:  %s\n"
+	       "    round count:  %u\n"
+	       "    ignore count: %u\n"
+	       "    cache size:   %" PRIi64 "\n"
+	       "    burst pattern:\n", config->policy == SINGLE ? "shared" : "per-worker",
```

Comment:
The  `write_uarea` size could be also printed. Currently user doesn't know if the value was limited by `max_uarea_size` capa.

Reply:
Good point, added in v5.